test/TestIdentity.js

```diff
@@ -0,0 +1,571 @@
+			await token.balanceOf(relayerAddr),
+			initialRelayerBal.toNumber() + feeAmount,
+			'relayer balance is correct'
+		)
+	})
+
+	// This is a "super-counterfactual" test: we will create an account, this account will earn from a channel,
+	// and THEN we will, at once, create the contract, sweep the channel and allow the account to withdraw their funds
+	it('IdentityFactory: deployAndExecute: create a new account after it has already earned from channels, withdraw via routines', async function() {
+		const tokenAmnt = 500
+		await token.setBalanceTo(userAcc, tokenAmnt * 3)
+
+		const txFeeAmount = 100
+
+		const [, initialFactoryBal] = await Promise.all([
+			token.balanceOf(relayerAddr),
+			token.balanceOf(identityFactory.address)
+		])
+
+		const maxToWithdraw = tokenAmnt - txFeeAmount
+		const accountToWithdrawTo = accounts[9]
+
+		// Create a new channel
+		const channel = sampleChannel(leader, follower, userAcc, token.address, 122)
+		// const signer = web3Provider.getSigner(relayerAddr)
+
+		const core = new Contract(coreAddr, Outpace._json.abi, web3Provider.getSigner(userAcc))
+		await (await core.deposit(channel.toSolidityTuple(), userAcc, tokenAmnt)).wait()
+
+		// Create a new account
+		const [bytecode, salt, expectedAddr] = createAccount([[userAcc, 1]], {
+			...getStorageSlotsFromArtifact(Identity)
+		})
+
+		// Prepare all the data needed for withdrawal
+		const [stateRoot, vsig1, vsig2, proof] = await getWithdrawData(channel, expectedAddr, tokenAmnt)
+		const withdrawal = new Withdraw({
+			channel,
+			balanceTreeAmount: tokenAmnt,
+			stateRoot,
+			sigLeader: vsig1,
+			sigFollower: vsig2,
+			proof
+		})
+
+		// Make the tx that will call the rotuines
+		const withdrawTx = new Transaction({
+			identityContract: expectedAddr,
+			nonce: 0,
+			feeTokenAddr: token.address,
+			feeAmount: 0,
+			to: coreAddr,
+			data: coreInterface.functions.withdraw.encode([withdrawal.toSolidityTuple()])
+		})
+
+		// Now the regular tx to withdraw our funds out of our Identity
+		const tokenInterface = new Interface(MockToken._json.abi)
+		const txToWithdraw = new Transaction({
+			identityContract: expectedAddr,
+			nonce: 1,
+			feeTokenAddr: token.address,
+			feeAmount: txFeeAmount,
+			to: token.address,
+			data: tokenInterface.functions.transfer.encode([accountToWithdrawTo, maxToWithdraw])
+		})
+		const txns = [withdrawTx, txToWithdraw]
+		const sigs = await Promise.all(
+			txns.map(async tx => splitSig(await ethSign(tx.hashHex(), userAcc)))
+		)
+
+		const handle = await identityFactory.deployAndExecute(
+			bytecode,
+			salt,
+			txns.map(tx => tx.toSolidityTuple()),
+			sigs,
+			{ gasLimit }
+		)
+
+		const receipt = await handle.wait()
+		assert.ok(receipt.events.some(x => x.event === 'LogDeployed'))
+		// LogDeployed, LogChannelWithdraw, channel sweep Transfer, 1 fee Transfers, the withdraw Transfer
+		assert.equal(receipt.events.length, 5, 'events length is right')
+		// console.log('gas used:', receipt.gasUsed.toNumber())
```

Comment:
debug? we can remove it

Reply:
it's to log gas used, it's not for debugging. it's helpful but it's commented out to not clog the test output